include/irr/asset/filters/CBlitImageFilter.h

```diff
@@ -174,13 +174,26 @@ class CBlitImageFilter : public CImageFilter<CBlitImageFilter<KernelX,KernelX,Ke
 						uint32_t		outLayerCount;
 					};
 				};
+
+				union NormalizeValues
+				{ 
+					NormalizeValues() {}
+					~NormalizeValues() {}
+
+					core::vectorSIMDu32 f; 
+					core::vectorSIMDu32 u;
+					core::vectorSIMDi32 i;
+				} oldMaxValue, oldMinValue;
+				
 				uint32_t				inMipLevel = 0u;
 				uint32_t				outMipLevel = 0u;
 				ICPUImage*				inImage = nullptr;
 				ICPUImage*				outImage = nullptr;
 				KernelX					kernelX;
 				KernelY					kernelY;
 				KernelZ					kernelZ;
+				asset::DefaultSwizzle defaultSwizzle;
```

Comment:
the swizzle type should be a template, because there are compile time swizzles and runtime polymorphic swizzles (see our other APIs)

Reply:
also you need an input swizzle and an output swizzle

Reply:
I don't get, we do i need in swizzle and out swizzle if filter producent only one out image
How shoould i process in?

Reply:
Blit takes an input image and puts it in the output image, while applying convolution kernels (or other kernels)

If you can swizzle, then you should be able to swizzle before and after (after decode and before encode) the kernel.